test/Server_test.fs

```diff
@@ -0,0 +1,200 @@
+module Async_rpc.Test.Server_test
+
+open System
+open System.Net
+open System.Net.Sockets
+open System.Threading.Tasks
+open Bin_prot
+open Core_kernel
+open Async_rpc
+open Async_rpc.Protocol
+open System.Threading.Tasks
+open System.Runtime.CompilerServices
+open NUnit.Framework
+open System.Threading
+
+let start_client_with_implementations port implementation_list =
+  let dispatch_tcs = new TaskCompletionSource<Or_error.t<string>>()
+  let client = new TcpClient("127.0.0.1", port)
+  let stream = client.GetStream() :> System.IO.Stream
+  let time = new Time_source.Wall_clock.t ()
+  let wait = new Auto_reset_event.t false
+
+  let connection_callback =
+    (fun connection ->
+      match connection with
+      | Error _error -> failwith "Connection failed"
+      | Ok connections ->
+        let Rpc =
+          Async_rpc.Rpc.create
+            {| name = "Rpc"; version = 9 |}
+            Type_class.bin_string
+            Type_class.bin_string
+
+        match
+          Async_rpc.Rpc.dispatch Rpc connections "Hello" (fun result ->
+            match result with
+            | Error error ->
+              dispatch_tcs.SetResult(Error(Protocol.Rpc_error.to_error (error)))
+            | Ok ok -> dispatch_tcs.SetResult(Ok ok)
+
+            Auto_reset_event.set wait)
+          with
+        | Error _error -> failwith "Dispatch failed"
+        | Ok () -> ()
+
+      ())
+
+  task {
+    Async_rpc.Connection.create
```

Comment:
I think this is already wrapped in a task internally so we might not need this one?